edgepi-thermocouple.js

```diff
@@ -1,43 +1,52 @@
-module.exports = function(RED) {
-    const rpc = require("@edgepi-cloud/edgepi-rpc")
+module.exports = function (RED) {
+  const rpc = require("@edgepi-cloud/edgepi-rpc");
 
-    function ThermocoupleNode(config) {
-        // Create node with user configs
-        RED.nodes.createNode(this, config);
-        const node = this;
-        const ipc_transport = "ipc:///tmp/edgepi.pipe"
-        const tcp_transport = `tcp://${config.tcpAddress}:${config.tcpPort}`
-        const transport = (config.transport === "Network") ? tcp_transport : ipc_transport;
+  const initializeNode = (config) => {
+    const transport =
+      config.transport === "Network"
+        ? `tcp://${config.tcpAddress}:${config.tcpPort}`
+        : "ipc:///tmp/edgepi.pipe";
 
-        // Init new tc
-        const tc = new rpc.TcService(transport)
-        if (tc){
-            console.debug("Thermocouple node initialized on: ", transport);
-            node.status({fill:"green", shape:"ring", text:"tc initialized"});
-          }
+    try {
+      const tc = new rpc.TcService(transport);
+      console.debug("Thermocouple node initialized on: ", transport);
+      node.status({ fill: "green", shape: "ring", text: "tc initialized" });
+      return tc;
+    } catch (error) {
+      console.error(error);
+      node.status({
+        fill: "red",
+        shape: "ring",
+        text: "Initialization error.",
+      });
+    }
+  };
 
-        // Input event listener
-        this.on('input', async function (msg, send, done) {
-            node.status({fill:"green", shape:"dot", text:"input recieved"});
-            try{
-                let temps = await tc.singleSample();
-                if(config.output === 'cj-lin'){
-                    msg.payload = temps
-                }
-                else{
-                    msg.payload = (config.output === 'cj') ? temps[0] : temps[1]
-                } 
-            }
-            catch(err) {
-                console.error(err);
-                msg.payload = err;
-            }
+  const ThermocoupleNode = (config) => {
+    RED.nodes.createNode(this, config);
+    const node = this;
+    let output = config.output;
 
-            send(msg);
-            if (done) {
-                done();
-            }
-        })
-    }
-    RED.nodes.registerType("thermocouple", ThermocoupleNode);
-}
+    initializeNode(config).then((tc) => {
+      this.on("input", async function (msg, send, done) {
+        node.status({ fill: "green", shape: "dot", text: "input recieved" });
+        try {
+          output = msg.payload || output;
+          let temps = await tc.singleSample();
+          if (output === "cj-lin") {
+            msg.payload = temps;
+          } else {
+            msg.payload = output === "cj" ? temps[0] : temps[1];
+          }
```

Comment:
we should only write  linearized temp to payload.

Reply:
DOne